### PR TITLE
Imp: short-circut checking validity of token

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem
When users had bunch of CE clusters saved, we checked each and every one when asking for particular workflow

# This PR's solution
Short-circut - quickly check if token for particular runtime is valid before sending out the request. 

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: projects/orquestra-sdk/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 🚨 *Breaking Changes*
 
+* Installing `orquestra-sdk[ray]` no longer installs Ray dashboard. Use `orquestra-sdk[all]` or `orquestra-sdk[dashboard]` instead.
```
Comment (SebastianMorawiec) [inline projects/orquestra-sdk/CHANGELOG.md:7]: Just a note: I forgot to put it here in my previous PR. Adding it as a drive-by fix

**Review by BenjaminMummery — APPROVED** [review]
Nitpick: The following files need their copyright updating:
- projects/orquestra-sdk/src/orquestra/sdk/_client/_base/_factory.py
- projects/orquestra-sdk/src/orquestra/sdk/_client/_base/_driver/_ce_runtime.py
- projects/orquestra-sdk/tests/sdk/test_consistent_return_shapes.py
- projects/orquestra-sdk/tests/sdk/driver/test_ce_runtime.py
- projects/orquestra-sdk/tests/sdk/driver/conftest.py

File: projects/orquestra-sdk/CHANGELOG.md (hunk 1 above)
Comment (alexjuda) [reply projects/orquestra-sdk/CHANGELOG.md:7]: In the past, we considered making automated changelog based on PR titles, and concluded that it's better to have a separate, hand-crafted file. IIRC, the reasoning was:
* it doesn't make senses for every PR to be part of the release notes
* the context is different. PR titles tend to be tailored for easy PR searching in history, or for onboarding code reviewers. The changelog uses a much different style, and often includes usage snippets
* it was difficult to enforce the rule of "PR title should read good as a changelog entry", especially when handling PRs made from infrequent, external contributors

I believe most of this reasoning also applies to commit messages, so I'd stick with hand-crafted release notes.

Comment (alexjuda) [reply projects/orquestra-sdk/CHANGELOG.md:7]: However! I'd be curious to explore automated version bumps based on commit messages on `main`. We squash branch content when we merge PRs, so this would mean relying on PR titles for inferring appropriate semver bump in the next release.
